Update to EFI Boot Guard v0.17

hoinmic opened the pull request.

hoinmic committed:
Update to EFI Boot Guard v0.17

petermarko commented on README.md:
```diff
@@ -26,6 +26,7 @@ The layer on master branch is compatible with the following Yocto version:
 
 Version  | Codename  | LTS
 -------- | --------- | ---
+5.0      | Scarthgap | Yes
```
This would need update of layer.conf.

I think we should do:
* test if new efibootguard is buildable with all yocto versions listed in layer.conf
* update layer.conf accordingly (add scarthgap, drop old releases which don't build anymore)

hoinmic replied:
The list is out of date. Thanks for the hint.

petermarko replied:
Ideally you update the layers/docs in separate commit.

Alternatively you add at least commit description describing the layer compatibility changes.

hoinmic replied:
done

petermarko commented on conf/layer.conf:
```diff
@@ -17,4 +17,4 @@ BBFILE_COLLECTIONS += "efibootguard"
 BBFILE_PRIORITY_efibootguard = "6"
 
-LAYERSERIES_COMPAT_efibootguard = "hardknott honister kirkstone"
+LAYERSERIES_COMPAT_efibootguard = "scarthgap kirkstone"
```
usual sorting order is oldest first
```suggestion
LAYERSERIES_COMPAT_efibootguard = "kirkstone scarthgap"
```

hoinmic replied:
ok

petermarko commented on README.md:
```diff
@@ -26,6 +26,7 @@ The layer on master branch is compatible with the following Yocto version:
+5.0      | Scarthgap | Yes
 4.0      | Kirkstone | Yes
 3.4      | Honister  | No
 3.3      | Hardknott | No
```
these two need to be removed from docs as you removed the layer compatibility with them

hoinmic replied:
done

hoinmic committed:
add scarthgap to compatible yocto versions

Adding scarthgap (yocto version 5.0) compatibility.
At the same time, old versions that are no longer
compatible have been removed.
